### ISSUE TITLE
wrong element?

### ISSUE THREAD

**Issue by hainm** [ParmEd/ParmEd#693]
Issue 1
======

What does `element` mean in ParmEd? I got '0' for Carbon

```python
import parmed as pmd

print(pmd.__version__)

parm = pmd.load_file("08Z.mol2", structure=True)
print(parm.atoms[0])
print(parm.atoms[0].element)
```

out
```bash
2.5.0+29.g172d95f
<Atom C1 [0]; In 08Z 0>
0
```

File: https://github.com/hainm/amber_things/tree/master/things/parmed/element

**updated**
Issue 2
======
should atom.element be something like "carbon" rather a number?

**Comment by hainm** [comment]
pytraj

```python
In [1]: traj = pt.load("./08Z.mol2")

In [2]: traj.top.atomlist[0].element
Out[2]: ('CARBON', 'C', 12.0107)
```

**Comment by swails** [comment]
Turns out that mol2 files don't store this information, and ParmEd never tries to assign it.  I should fix that and at least make it as smart as the GRO file's element assignment.

**Comment by hainm** [comment]
so `element` is `element` for atom? 

should it be something like `carbon` rather a number? 

```python
In [1]: parm = pmd.load_file('tz2.parm7')

In [2]: parm.atoms[0].element
Out[2]: 7

In [3]: parm.atoms[0]
Out[3]: <Atom N [0]; In SER 0>

In [4]: parm.atoms[0].atomic_number
Out[4]: 7

```

**Comment by ctk3b** [comment]
cc #562

**Comment by swails** [comment]
Wow!  I usually remember threads like that! Thanks for digging that up.

Maybe worth revisiting.  The behavior that I just added has a little more chemistry-awareness -- it does what was done for the GRO file (which is a *lot* improved since #668) using the atom names, but is smart enough to assign ions to metal elements if the names match there (which are determined as those atoms which aren't bonded to anybody else).

This would be helpful in doing file conversions to OFF files or other atomic number-bearing formats, but good concerns were raised in #562.  Should I just close this one, too?

**Comment by ctk3b** [comment]
@davidlmobley would probably want to take a look at it :)

**Comment by hainm** [comment]
@swails 

there is still open question

`should atom.element be something like "carbon" rather a number? `

or you can close and I open new issue.

**Comment by swails** [comment]
> should atom.element be something like "carbon" rather a number?

Unfortunately I made `element` an alias to `atomic_number` awhile ago (this is part of what made me come to believe aliases were bad!)

I might change `atom.element` at some point, but not until I rework `parmed.periodic_table` to actually have element objects, so it would be `parmed.periodic_table.carbon`, not "carbon".  But that's low low on the priority list (and still might not happen).

**Comment by hainm** [comment]
turns out that I asked and you answered before: https://github.com/ParmEd/ParmEd/issues/266
